test/studies/IMSuite/vertexColoring.chpl

```diff
@@ -0,0 +1,323 @@
+ /* This file is part of IMSuite Benchamark Suite.
+
+    This file is licensed to You under the Eclipse Public License (EPL);
+    You may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.opensource.org/licenses/eclipse-1.0.php
+
+   (C) Copyright IMSuite 2013-present.
+ */
+
+ /*
+    vertexColoring aims to color the vertices of a tree with
+    three colors.
+
+    @author Suyash Gupta
+    @author V Krishna Nandivada
+
+    Ported By: Kushal Singh
+ */
+
+module vertexColoring {
+
+    use Random;
+    use VisualDebug;
+
+    config var inputFile = "inputvertexColoring_16_max.txt";
+    config var outputFile = "outputvertexColoring.txt";
+    config var verify = false;
+
+    var nodes: int;
+    var colorLabel: int;
+    var D : domain(1);
+    var adj_graph :[D][D] int;
```

Comment:
Should this be an array over a 2D domain instead of an array-of-arrays? For example, you can write something like this in chapel:
```chpl
var dom = {1..5, 1..5};
var data : [dom] int;
```
We generally prefer this method when possible, and it usually leads to more elegant code.